src/EntityFramework/Core/Query/PlanCompiler/GroupAggregateRefComputingVisitor.cs

```diff
@@ -154,20 +154,35 @@ public override void Visit(FunctionOp op, Node n)
             {
                 return;
             }
-            PlanCompiler.Assert(n.Children.Count == 1, "Aggregate Function must have one argument");
-
-            var argumentNode = n.Child0;
-
             GroupAggregateVarInfo referencedGroupAggregateVarInfo;
+            GroupAggregateVarInfo referencedGroupAggregateVarInfoTracker = null;
+
             Node templateNode;
             bool isUnnested;
-            if (GroupAggregateVarComputationTranslator.TryTranslateOverGroupAggregateVar(
-                n.Child0, false, _command, _groupAggregateVarInfoManager, out referencedGroupAggregateVarInfo, out templateNode,
-                out isUnnested)
-                &&
-                (isUnnested || AggregatePushdownUtil.IsVarRefOverGivenVar(templateNode, referencedGroupAggregateVarInfo.GroupAggregateVar)))
+
+            var list = new List<Node>();
+
+            foreach (var argument in n.Children)
+            {
+                if (GroupAggregateVarComputationTranslator.TryTranslateOverGroupAggregateVar(
+                    argument, false, _command, _groupAggregateVarInfoManager, out referencedGroupAggregateVarInfo, out templateNode,
+                    out isUnnested)
+                    &&
+                    (isUnnested || AggregatePushdownUtil.IsVarRefOverGivenVar(templateNode, referencedGroupAggregateVarInfo.GroupAggregateVar)))
+                {
+
+                    referencedGroupAggregateVarInfoTracker = referencedGroupAggregateVarInfo;
```

Comment:
if two (or more) arguments meet this condition, we will only add CandidateAggregateNodes for the second (last) one. This doesn't seem to be the right behavior. It might be safer to just disable this optimization for aggregate functions with multiple arguments, just to be safe.